src/VisualStudio/Core/Impl/CodeModel/AbstractCodeModelService.cs

```diff
@@ -488,30 +488,15 @@ protected EnvDTE80.CodeAttributeArgument CreateInternalCodeAttributeArgument(Cod
 
         public abstract string GetFullyQualifiedName(string name, int position, SemanticModel semanticModel);
 
-        public void Rename(ISymbol symbol, string newName, Solution solution)
+        public void Rename(ISymbol symbol, string newName, Workspace workspace, ProjectCodeModelFactory projectCodeModelFactory)
         {
-            // TODO: (tomescht) make this testable through unit tests.
-            // Right now we have to go through the project system to find all
-            // the outstanding CodeElements which might be affected by the
-            // rename. This is silly. This functionality should be moved down
-            // into the service layer.
-
-            var workspace = solution.Workspace as VisualStudioWorkspace;
```

Comment:
I'm not including this check; it's unclear to me how somebody would get a CodeElement not backed by a VisualStudioWorkspace to begin with. And also nothing here actually used the VisualStudioWorkspace-ness of it in any way.

Reply:
Fair enough